protocol/parser/parser.go

```diff
@@ -307,7 +307,7 @@ func ParseDictionary(rpcInput RPCInput, input []string, dataSource int) ([]inter
 	case map[string]interface{}:
 		// If attribute with key propName exists return value
 		if val, ok := unmarshalledDataTyped[propName]; ok {
-			return appendInterfaceToInterfaceArray(val), nil
+			return appendInterfaceToInterfaceArray(fmt.Sprintf("%v", val)), nil
```

Comment:
this is not the right way to do it, if grpc has objects unmarshalled not as strings which is why I assume you need it changed you can have a method to check the type and convert only the necessary and known types
```
convertInterfaceToString(input interface{}) (string, error) {
switch inp := input .(type) {
	case string:
		return info
	case int:
	       return strconv.formatInt(info,10)
	default:
	       return utils.LavaFormatError("Failed to convert to string, unknown type", info: input , inp  etc..)
}
```

Reply:
that way we know which types we support and which we do not

Reply:
Yes the reason was grpc being unmarshalled not as strings. Fixed here: c261e1ecef87053813dee7dc1761a4ff6fe5b86c